### ISSUE TITLE
Typed dict JSON schema does not pay attention to `extra_behavior` argument

### ISSUE THREAD

**Issue by me-and** [pydantic/pydantic#7809]
### Initial Checks

- [X] I confirm that I'm using Pydantic V2

### Description

When creating a JSON schema from a typed dict core schema, the core schema uses the attribute `extra_behavior` to determine whether additional unspecified fields are allowed. However the code for generating JSON schemas, at https://github.com/pydantic/pydantic/blob/v2.4.2/pydantic/json_schema.py#L1194, instead checks for the key `extra`, so the setting is ignored.

### Example Code

```Python
from pydantic.json_schema import GenerateJsonSchema
import pydantic.version
import pydantic_core.core_schema as core_schema

schema = core_schema.typed_dict_schema(
        {'a': core_schema.typed_dict_field(core_schema.str_schema())},
        extra_behavior='forbid',
        )

json_schema = GenerateJsonSchema().generate(schema)

try:
    assert json_schema['additionalProperties'] == False
except (KeyError, AssertionError):
    print('Generated JSON schema missing correct extra_behavior output')
    print('Input schema:')
    print(repr(schema))
    print('Generated schema:')
    print(repr(json_schema))
    print('Version info:')
    print(pydantic.version.version_info())
```

### Python, Pydantic & OS Version

```Text
pydantic version: 2.4.2
        pydantic-core version: 2.10.1
          pydantic-core build: profile=release pgo=false
                 install path: /home/adam/.local/lib/python3.11/site-packages/pydantic
               python version: 3.11.2 (main, Mar 13 2023, 12:18:29) [GCC 12.2.0]
                     platform: Linux-6.1.21-v8+-aarch64-with-glibc2.36
             related packages: typing_extensions-4.8.0 email-validator-2.0.0.post2 typing_extensions-4.4.0 mypy-1.0.1
```

**Comment by me-and** [comment]
Hmm. My naive fix at #7810 might not be sufficient here; this is complicated by the current setup using `extra` in ConfigDict to mean the same thing, so it might be that this wants to be fixed by changing pydantic_core to expect an `extra` argument here, instead of or as well as the `extra_behavior` argument…

**Comment by sydney-runkle** [comment]
@me-and,

Thanks for reporting this and for the contribution! I'll get back to you on Monday with a more detailed review of your PR 😄.
